pkg/kubelet/dockertools/manager.go

```diff
@@ -1354,6 +1354,10 @@ func (dm *DockerManager) killContainer(containerID kubecontainer.ContainerID, co
 	if gracePeriod < minimumGracePeriodInSeconds {
 		gracePeriod = minimumGracePeriodInSeconds
 	}
+	if gracePeriodOverride != nil {
```

Comment:
@smarterclayton - 2s is too long when running out of resources.  Are you asking why do we have a minimum at all in the normal case?